src/integrationTest/java/org/opensearch/security/FlsDlsAndFieldMaskingTest.java

```diff
@@ -0,0 +1,953 @@
+					.alias(FIRST_INDEX_ALIAS)
+			)).actionGet();
+			client.admin().indices().aliases(new IndicesAliasesRequest().addAliasAction(new IndicesAliasesRequest.AliasActions(ADD)
+					.index(FIRST_INDEX_NAME)
+					.alias(FIRST_INDEX_ALIAS_FILTERED_BY_NEXT_SONG_TITLE)
+					.filter(QueryBuilders.queryStringQuery(QUERY_TITLE_NEXT_SONG))
+			)).actionGet();
+			client.admin().indices().aliases(new IndicesAliasesRequest().addAliasAction(new IndicesAliasesRequest.AliasActions(ADD)
+					.index(FIRST_INDEX_NAME)
+					.alias(FIRST_INDEX_ALIAS_FILTERED_BY_TWINS_ARTIST)
+					.filter(QueryBuilders.queryStringQuery(String.format("%s:%s", FIELD_ARTIST, ARTIST_TWINS)))
+			)).actionGet();
+			client.admin().indices().aliases(new IndicesAliasesRequest().addAliasAction(new IndicesAliasesRequest.AliasActions(ADD)
+					.index(FIRST_INDEX_NAME)
+					.alias(FIRST_INDEX_ALIAS_FILTERED_BY_FIRST_ARTIST)
+					.filter(QueryBuilders.queryStringQuery(String.format("%s:%s", FIELD_ARTIST, ARTIST_FIRST)))
+			)).actionGet();
+
+			SECOND_INDEX_SONGS_BY_ID.forEach((id, song) -> {
+				client.prepareIndex(SECOND_INDEX_NAME).setId(id).setRefreshPolicy(IMMEDIATE).setSource(song.asMap()).get();
+			});
+			client.admin().indices().aliases(new IndicesAliasesRequest().addAliasAction(new IndicesAliasesRequest.AliasActions(ADD)
+					.indices(SECOND_INDEX_NAME)
+					.alias(SECOND_INDEX_ALIAS)
+			)).actionGet();
+		}
+	}
+
+	@Test
+	public void flsEnabledFieldsAreHiddenForNormalUsers() throws IOException {
+		String indexName = "fls_index";
+		String indexAlias = "fls_index_alias";
+		String indexFilteredAlias = "fls_index_filtered_alias";
+		TestSecurityConfig.Role userRole = new TestSecurityConfig.Role("fls_exclude_stars_reader")
+				.clusterPermissions("cluster_composite_ops_ro")
+				.indexPermissions("read")
+				.fls("~".concat(FIELD_STARS))
+				.on("*");
+		TestSecurityConfig.User user = createUserWithRole("fls_user", userRole);
+		List<String> docIds = createIndexWithDocs(indexName, SONGS[0], SONGS[1]);
+		addAliasToIndex(indexName, indexAlias);
+		addAliasToIndex(indexName, indexFilteredAlias, QueryBuilders.queryStringQuery(String.format("%s:%s", FIELD_ARTIST, SONGS[0].getArtist())));
+
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(user)) {
+			//search
+			SearchRequest searchRequest = new SearchRequest(indexName);
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertSearchHitsDoNotContainField(searchResponse, FIELD_STARS);
+
+			//search with index pattern
+			searchRequest = new SearchRequest("*".concat(indexName));
+			searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertSearchHitsDoNotContainField(searchResponse, FIELD_STARS);
+
+			//search via alias
+			searchRequest = new SearchRequest(indexAlias);
+			searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertSearchHitsDoNotContainField(searchResponse, FIELD_STARS);
```

Comment:
I think these reuses of request/response are a good case for a loop, can we generate a list of request and then iterate through them repeating the same verification?


Maybe something like the following would be easy to generate all the index patterns?  Alternatively if `new SearchRequest(...)` is easy as a one liner that would be easy to read/maintain
```java
   List<String> indicesToCheck = List.of(
      "*".concat(indexName),
      indexAlias,
      ...);
   // Loop over request/response
```

Reply:
fixed

Reply:
Thank you for making this update, much cleaner - can we do this to all the tests in this PR?